DevelopersGuide.md

```diff
@@ -0,0 +1,192 @@
+openshift/oauth-proxy:$OAUTH_VERSION
+openshift/origin-sti-builder:$OC_VERSION
+openshift/origin-deployer:$OC_VERSION
+openshift/origin-docker-registry:$OC_VERSION
+openshift/origin-haproxy-router:$OC_VERSION
+openshift/origin:$OC_VERSION
+openshift/origin-pod:$OC_VERSION
+```
+
+```bash
+DOCKER_REGISTRY_CACHE_FOLDER="/500GB/fabric8_vm_cache/docker_registry_cache"
+docker run -p 5000:5000 -d --restart=always \
+            --name registry_cache \
+            -e REGISTRY_PROXY_REMOTEURL=http://registry-1.docker.io \
+            -v $DOCKER_REGISTRY_CACHE_FOLDER:/var/lib/registry:Z \
+            registry:2
+```
+- to enable the local mirror in Minishift pass this flag: `--registry-mirror http://$(ip a s wlp4s0 | grep "inet "  | grep -oE "\b([0-9]{1,3}\.){3}[0-9]{1,3}\b" | head -1):5000` , where the complex expression is just a way to extract an ip address that the VM is able to reach. I use my Wifi assigned ones, but this depends on the complexity of your network configuration. At the end it's just a matter of providing something like: `--registry-mirror http://192.168.1.2:5000`
+
+##### Install companion script
+- there is an useful management scripts hiddeen in `syndesis-ui` repo: https://github.com/syndesisio/syndesis-ui/blob/master/scripts/syndesis-install providing a set of options around the usual steps defined in the quickstarts to bring up an env
+- useful options are:
+  - `--pull` to download all the images you need
+  - `--clean` to delete the `project` at Openshift level (so that you can re-run a `oc new-app`)
+  - `--from` to load Openshift definition files from a local folder you specify
+  - `--namespace` to change the namespace to install Syndesis into
+  - `--hostname` to specify an hostname for your vm
+
+##### Bring up minishift console
+- `minishift console` or `minishift status`
+
+## Work on a localy copy of syndesis-rest
+- from `syndesis-rest/README.md`:
+```bash
+# create image
+eval $(minishift docker-env)
+cd runtime
+# build code and produce corresponding docker images
+mvn fabric8:build -Dfabric8.mode=kubernetes
+
+# now redploy the rest pod (either from cli or from web)
+oc rollout latest dc/syndesis-rest
+```
+
+##### Collection of helper scripts to automaticlaly build locally and deploy to minishift
+- they build AND deploy
+- in `syndesis-project/tools/minishift-build/`:
+```bash
+atlasmap.sh
+build-all.sh
+integration-runtime.sh
+README.md
+rest.sh
+ui.sh
+vars.sh
+verifier.sh
+```
+
+
+
+## Web development
+- based on the idea of modify a service (route) definition in Syndesis Openshift config, so that it will route the invocation to the local dev webeserver running on host
+- this happens in this script: `scripts/minishift-setup.sh`
+0. install `yarn`
+```bash
+sudo dnf config-manager --add-repo https://dl.yarnpkg.com/rpm/yarn.repo
+sudo dnf install -y yarn
+```
+1. build `syndesis-ui`
+```
+cd syndesis-ui
+yarn install
+cd scripts
+sh minishift-setup.sh
+sh syndesis-install
```

Comment:
Switch these two commands around, i.e. `syndesis-install` and then `minishift-setup.sh`

Technically we could even consider adding yarn targets for both scripts, like `yarn setup:minishift`